### PR TITLE
Feature - add OpenApi documentation for the Correlation headers

### LINKED ISSUE #155: Document correlation headers in OpenAPI specs, when applicable
**Is your feature request related to a problem? Please describe.**
Document correlation headers in OpenAPI specs, when applicable

**Describe the solution you'd like**
Use [`Swashbuckle.AspNetCore.Filters`](https://www.nuget.org/packages/Swashbuckle.AspNetCore.Filters/).

We need to:
- Configure it in `AddSwaggerGen` via `AddHeaderOperationFilter` & `AddResponseHeadersFilter` (see [docs](https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters#installation))
- Annotate our health endpoint with `SwaggerResponseHeader` (see [docs](https://github.com/mattfrear/Swashbuckle.AspNetCore.Filters#how-to-use---response-headers))

**Describe alternatives you've considered**
None

**Additional context**
This should only happen when `exclude-correlation` & `exclude-openApi` are `false`

### PR DESCRIPTION
**Pull request by stijnmoreels**
Closes #155

### CODE REVIEW

File: src/Arcus.Templates.WebApi/Controllers/HealthController.cs [hunk 1]
```diff
@@ -35,6 +38,10 @@ public HealthController(HealthCheckService healthCheckService)
         [HttpGet(Name = "Health_Get")]
         [ProducesResponseType(typeof(HealthReport), StatusCodes.Status200OK)]
         [ProducesResponseType(typeof(HealthReport), StatusCodes.Status503ServiceUnavailable)]
+#if (ExcludeCorrelation == false && ExcludeCorrelation == false)
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Controllers/HealthController.cs:41]: This looks off, maybe use this instead?
```suggestion
#if (ExcludeOpenApi == false && ExcludeCorrelation == false)
```

File: src/Arcus.Templates.WebApi/Controllers/HealthController.cs [hunk 2]
```diff
@@ -3,6 +3,9 @@
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 using GuardNet;
+#if (ExcludeCorrelation == false && ExcludeCorrelation == false)
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Controllers/HealthController.cs:6]: This looks off, maybe use this instead?
```suggestion
#if (ExcludeOpenApi == false && ExcludeCorrelation == false)
```

File: src/Arcus.Templates.WebApi/Startup.cs [hunk 3]
```diff
@@ -158,6 +161,12 @@ public void ConfigureServices(IServiceCollection services)
             {
                 swaggerGenerationOptions.SwaggerDoc("v1", openApiInformation);
                 swaggerGenerationOptions.IncludeXmlComments(Path.Combine(AppContext.BaseDirectory, "Arcus.Templates.WebApi.Open-Api.xml"));
+
+#if (ExcludeCorrelation == false)
+                swaggerGenerationOptions.OperationFilter<AddHeaderOperationFilter>("RequestId", "The header that will have an unique identifier to distinguish between requests/responses", false);
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Startup.cs:166]: Request Id should not be in the request header docs
```suggestion
```

File: src/Arcus.Templates.WebApi/Controllers/HealthController.cs [hunk 4]
```diff
@@ -35,6 +38,10 @@ public HealthController(HealthCheckService healthCheckService)
         [HttpGet(Name = "Health_Get")]
         [ProducesResponseType(typeof(HealthReport), StatusCodes.Status200OK)]
         [ProducesResponseType(typeof(HealthReport), StatusCodes.Status503ServiceUnavailable)]
+#if (ExcludeCorrelation == false && ExcludeCorrelation == false)
+        [SwaggerResponseHeader(200, "RequestId", "string", "The header that will have an unique identifier to distinguish between requests/responses")]
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Controllers/HealthController.cs:42]: What about this? This feels more related to what it actually does, no?
```suggestion
        [SwaggerResponseHeader(200, "RequestId", "string", "The header that has a request ID that uniquely identifies this operation call")]
```

File: src/Arcus.Templates.WebApi/Controllers/HealthController.cs [hunk 5]
```diff
@@ -35,6 +38,10 @@ public HealthController(HealthCheckService healthCheckService)
...
+        [SwaggerResponseHeader(200, "X-Transaction-Id", "string", "The header that will have an identifier that relate requests/responses")]
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Controllers/HealthController.cs:43]: What about this, isn't this more clear?
```suggestion
        [SwaggerResponseHeader(200, "X-Transaction-Id", "string", "The header that has the transaction ID is used to correlate multiple operation calls.")]
```

File: src/Arcus.Templates.WebApi/Startup.cs [hunk 6]
```diff
@@ -158,6 +161,12 @@ public void ConfigureServices(IServiceCollection services)
...
+                swaggerGenerationOptions.OperationFilter<AddHeaderOperationFilter>("X-Transaction-Id", "The header that will have an identifier that relate requests/responses", false);
```
Comment (tomkerkhove) [inline src/Arcus.Templates.WebApi/Startup.cs:167]: What about this? Feels more clear, no?
```suggestion
                swaggerGenerationOptions.OperationFilter<AddHeaderOperationFilter>("X-Transaction-Id", "Transaction ID is used to correlate multiple operation calls. A new transaction ID will be generated if not specified.", false);
```

File: src/Arcus.Templates.WebApi/Controllers/HealthController.cs (hunk 4 above)
Comment (stijnmoreels) [reply src/Arcus.Templates.WebApi/Controllers/HealthController.cs:42]: Same for me.

**Pull request merged**